lib/checkpoint-writer.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+const EventEmitter = require('events');
+const _ = require('lodash');
+let writerLoopStopped = true;
+let lastDoc;
+let lastCheckpointId;
+let harvesterApp;
+let writeInterval = 1;
+const log = require('agco-logger')({});
```

Comment:
We can actually pull out agco-logger now, since we've learned all we will from profiling, and we can get rid of a ton of environment variables. You don't have to do this now though, we could do it some other time if you like, but it's prolonging technical debt.